test/extended/util/oauthserver/oauthserver.go

```diff
@@ -152,51 +143,87 @@ func DeployOAuthServer(oc *exutil.CLI, idps []osinv1.IdentityProvider, configMap
 		},
 	})
 	if err != nil {
-		return nil, cleanups, err
+		return nil, cleanupFunc, err
 	}
 
 	// get the OAuth server image that's used in the cluster
 	image, err := getImage(oc)
 	if err != nil {
-		return nil, cleanups, err
+		return nil, cleanupFunc, err
 	}
 
 	// prepare the pod def, create secrets and CMs
 	oauthServerPod, err := oauthServerPod(configMaps, secrets, image)
 	if err != nil {
-		return nil, cleanups, err
+		return nil, cleanupFunc, err
 	}
 
 	// finally create the oauth server, wait till it starts running
-	if _, err := coreClient.Pods(oc.Namespace()).Create(oauthServerPod); err != nil {
-		return nil, cleanups, err
+	if _, err := kubeClient.CoreV1().Pods(oc.Namespace()).Create(oauthServerPod); err != nil {
+		return nil, cleanupFunc, err
+	}
+
+	if err := waitForOAuthServerReady(oc); err != nil {
+		return nil, cleanupFunc, err
+	}
+
+	oauthClient, err := createOAuthClient(oc, routeURL)
+	if err != nil {
+		return nil, cleanupFunc, err
+	}
+	cleanupFuncs = append(cleanupFuncs, func() {
+		_ = oc.AsAdmin().Run("delete").Args("oauthclients.oauth.openshift.io", oauthClient.Name).Execute()
+	})
+
+	newRequestTokenOptionFunc := func(username, password string) *tokencmd.RequestTokenOptions {
+		return newRequestTokenOptions(restclient.AnonymousClientConfig(oc.AdminConfig()), routeURL, oc.Namespace(), username, password)
 	}
 
-	err = wait.PollImmediate(1*time.Second, 45*time.Second, func() (bool, error) {
+	return newRequestTokenOptionFunc, cleanupFunc, nil
+}
+
+func waitForOAuthServerReady(oc *exutil.CLI) error {
+	if err := exutil.WaitForUserBeAuthorized(oc, "system:serviceaccount:"+oc.Namespace()+":e2e-oauth", "get", "oauthclients.oauth.openshift.io"); err != nil {
```

Comment:
check for `*` `*` since that is what it is granted.

Reply:
Done